src/Nox.Generator/Infrastructure/Persistence/DtoDbContextGenerator.cs

```diff
@@ -22,115 +22,15 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
-                code.AppendLine();
-            }
-        }
-
-        AddDbContextOnConfiguring(code, codeGeneratorState);
-
-        AddOnModelCreating(code);
-
-        code.EndBlock();
-
-        code.GenerateSourceCode();
-    }
-
-    private static void AddOnModelCreating(CodeBuilder code)
-    {
-        code.AppendLine(@$" protected override void OnModelCreating(ModelBuilder modelBuilder)");
-        code.StartBlock();
-        code.AppendLine(@$"base.OnModelCreating(modelBuilder);");
-        code.AppendLine(@$"
-            if (_noxSolution.Domain != null)
-            {{
-                var codeGeneratorState = new NoxSolutionCodeGeneratorState(_noxSolution, _clientAssemblyProvider.ClientAssembly);
-                foreach (var entity in codeGeneratorState.Solution.Domain!.Entities)
-                {{
-                    // Ignore owned entities configuration as they are configured inside entity constructor
-                    if (entity.IsOwnedEntity)
-                    {{
-                        continue;
-                    }}
-
-                    var type = codeGeneratorState.GetEntityDtoType(entity.Name + ""Dto"");
-                    if (type != null)
-                    {{
-                       _noxDtoDatabaseConfigurator.ConfigureDto(codeGeneratorState, new Nox.Types.EntityFramework.EntityBuilderAdapter.EntityBuilderAdapter(modelBuilder.Entity(type)), entity);
-                    }}
-                    else
-                    {{
-                        throw new Exception($""Could not resolve type for {{entity.Name}}Dto"");
-                    }}
-                }}
-            }}            
-        ");
-        code.EndBlock();
-    }
-
-    internal static void AddDbContextOnConfiguring(CodeBuilder code, NoxSolutionCodeGeneratorState codeGeneratorState)
-    {
-        code.AppendLine("protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)");
-        code.StartBlock();
-        code.AppendLine("base.OnConfiguring(optionsBuilder);");
-        code.AppendLine("if (_noxSolution.Infrastructure is { Persistence.DatabaseServer: not null })");
-        code.StartBlock();
-        code.AppendLine($"_dbProvider.ConfigureDbContext(optionsBuilder, \"{codeGeneratorState.Solution.Name}\", _noxSolution.Infrastructure!.Persistence.DatabaseServer); ");
-        code.EndBlock();
-        code.EndBlock();
-        code.AppendLine();
+        const string className = $"DtoDbContext";
+        const string templateName = @"Infrastructure.Persistence.DtoDbContext";
+        var entities = solution.Domain.Entities.Where(e => !e.IsOwnedEntity).ToList();
+        
+        new TemplateCodeBuilder(context, codeGeneratorState)
+            .WithClassName(className)
+            .WithFileNamePrefix($"Infrastructure.Persistence")
+            .WithObject("RootNameSpace", codeGeneratorState.RootNameSpace)
```

Comment:
no need this object use codeGeneratorState.RootNameSpace directly on the template

Reply:
Desired changes were applied